Assets/Scripts/StartTrial.cs

```diff
@@ -7,13 +7,23 @@
 public class StartTrial : MonoBehaviour
 {
     public LoadTerrain terrain_bool;
-    public Session session;
-    void OnTriggerEnter()
+    
+    //public Session session;
+    void OnTriggerEnter(Collider other)
     {
-        if(!session.InTrial)
+        if(!Session.instance.InTrial)
         {
-            terrain_bool.is_mirrored = false;
-            session.BeginNextTrial();
+            if(other.transform.position.z < 0)
```

Comment:
not that you should, but I wonder if it were possible to have these `if` statements driven by some external piece of logic that changed a public variable to `True` or `False` depending on the state of some relevant thing (e.g. other versions of the current case where you're checking if `other`'s `z` value is + or -_

...that would allow you to make things appear and disappear based moer flexibly, e.g. you could make 'zones' or whatever based on the `x` and `z` values of markers, etc.